### ISSUE TITLE
Mobile iOS poor performance solved with CSS.

### ISSUE THREAD

**Issue by SamHall** [openseadragon/openseadragon#2434]
I spent hours chasing down what ended up being one line CSS - sharing here for others.

In collection mode, the same content would be jerky and slower on an iPhone in Chrome.
I finally narrowed down the solution, which has been posted here before - perhaps not specific to collection mode.
Is there a better 'baked in' solution for a future version?

The magic fix:
```
.openseadragon-canvas:focus {
	outline: 0 !important;
}
```

_Example WITHOUT css._
https://chattanoogahistory.com/collection_test_one.php

_Example WITH css._
https://chattanoogahistory.com/collection_test_two.php

**Comment by iangilman** [comment]
Yeah, I don't think it has to do with collection mode. Clearly we've discussed this issue before: https://github.com/openseadragon/openseadragon/issues/1408 https://github.com/openseadragon/openseadragon/issues/893 https://github.com/openseadragon/openseadragon/issues/1569 https://github.com/openseadragon/openseadragon/issues/1753 https://github.com/openseadragon/openseadragon/issues/1938. 

We don't want to just remove the focus outline because it's an accessibility feature. That said,  I don't think it's relevant on mobile, so maybe what we should do is detect mobile and remove it on that case.

Thank you for bringing it back to attention! I don't suppose you'd be interested in working on a patch?

**Comment by SamHall** [comment]
If I had skills in such, I'd gladly step up. However, I struggle to piece together a working modification to a script.

**Comment by iangilman** [comment]
@SamHall No worries! Thank you again for bringing it up :)

**Comment by kabeersiddiqui** [comment]
No worries! Thank you again for bringing it up
[https://www.mobilezmarket.com/](url)






























































[](url)

**Comment by Sumaiya2505** [comment]
Hi @iangilman , 
`@media (hover: none) and (pointer: coarse) {
  .openseadragon-canvas:focus {
    outline: none !important;
  }
}
`
I suppose that this would work fine for the touch devices.Whether the focus outline needs to removed on all the touch devices or only on iOS ?? Could please share your view on this ???

**Comment by iangilman** [comment]
Today I learned about the hover and pointer media queries! Very cool. 

I'm not sure we need the pointer media query in this case. If we did, maybe it should be `or` instead of `and`?

Anyway, OpenSeadragon doesn't come with a CSS stylesheet. Is there a way to include media queries on the page without an explicit stylesheet? At any rate, we want to be able to apply this via JavaScript without having to add a stylesheet to our deliverables.

And yes, I think we should remove the outline on all touch devices, not just iOS.

This seems like a great issue for you to start with!

**Comment by Sumaiya2505** [comment]
@iangilman  Thanks for the support.I will start working on this issue.

**Comment by iangilman** [comment]
@Sumaiya2505 Excellent. I look forward to the result!

**Comment by Sumaiya2505** [comment]
Hi @iangilman 
I have raised the PR to address this issue ,Please review and let me know the changes : #2434 https://github.com/openseadragon/openseadragon/pull/2732 
Thanks.

**Comment by Sumaiya2505** [comment]
Hi @iangilman,
Please review the PR raised with the changes requested and let me know if there are any issues.
Thanks

**Comment by iangilman** [comment]
Fixed by #2732
